synapse/rest/media/v1/preview_url_resource.py

```diff
@@ -426,8 +431,7 @@ def _expire_url_cache_data(self):
 
         yield self.store.delete_url_cache_media(removed_media)
 
-        if removed_media:
-            logger.info("Deleted %d media from url cache", len(removed_media))
+        logger.info("Deleted %d media from url cache", len(removed_media))
```

Comment:
Why have we removed the conditional? We added it so that we wouldn't spam the logs with lots of pointless log lines.

Reply:
Because it makes a nice closure to https://github.com/matrix-org/synapse/pull/2697/files#diff-3c2efbd259756c1c1dc5dbdf467b5570R355.

I don't think that two log lines every ten seconds constitutes a significant amount of spam. It's useful to know it's running; if nothing else it highlights the fact that it was running four times every ten seconds (leading to #2701).

Reply:
People specifically added a PR as it was annoying, fwiw. Once every ten seconds isn't too bad by itself, but we have quite a few background tasks that should probably be logging to the same extent. 

Personally, I'd probably put the first log line after we ask the DB for expired URLs and then log both if any expired media ids were returned or we're at debug logging.

Reply:
> People specifically added a PR as it was annoying, fwiw.

wat? people added a PR?

> we have quite a few background tasks that should probably be logging to the same extent.

Probably. I'm not going to fix them in this PR though.

> Personally, I'd probably put the first log line after we ask the DB for expired URLs and then log both if any expired media ids were returned or we're at debug logging.

I'm really unhappy about background tasks that run silently. Not least because it leads to the sort of fuck-up we had here where it's running much more often than we think it is and it's only good luck which stops the processes racing against each other and doing real damage.